### PR TITLE
Add an HTTPRequest usage example to the class reference

### PR DESCRIPTION
**Pull request by Calinou**
This also adds a link to the "Making HTTP requests" tutorial.

### CODE REVIEW

File: doc/classes/HTTPRequest.xml [hunk 1]
```diff
@@ -1,13 +1,43 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="HTTPRequest" inherits="Node" category="Core" version="3.2">
 	<brief_description>
-		A node with the ability to send HTTP requests.
+		A node with the ability to send HTTP(S) requests.
 	</brief_description>
 	<description>
 		A node with the ability to send HTTP requests. Uses [HTTPClient] internally.
```
Comment (aaronfranke) [inline doc/classes/HTTPRequest.xml]: If you're changing the brief description to "HTTP(S)" then perhaps this line should be "HTTP(S)" or perhaps "HTTP or HTTPS".

File: doc/classes/HTTPRequest.xml [hunk 2]
```diff
@@ -1,13 +1,43 @@
...
 		Can be used to make HTTP requests, i.e. download or upload files or web content via HTTP.
+		[b]Example of loading and displaying an image using HTTPRequest:[/b]
+		[codeblock]
+		func _ready():
+		    # Create an HTTP request node and connect its completion signal
```
Comment (akien-mga) [inline doc/classes/HTTPRequest.xml]: Unlike PEP-8, we also use a dot at the end of comments even with a single sentence.
This should likely be made clear in the GDScript style guide BTW if it isn't yet.

File: doc/classes/HTTPRequest.xml (hunk 1 above)
Comment (akien-mga) [reply doc/classes/HTTPRequest.xml]: I'm not an expert on HTTP, but when using HTTPS do we actual call those requests "HTTPS requests", or are they still "HTTP requests" that happen to use the secure protocol? (i.e., are those requests the same, or are they different and thus should be explicitly listed as two types of requests supported by this node)

**Comment by akien-mga** [comment]
Thanks!

File: doc/classes/HTTPRequest.xml (hunk 1 above)
Comment (dsge) [reply doc/classes/HTTPRequest.xml]: HTTPS requests are the same as HTTP requests. They just happen to use encryption while transferring data. 

You do not need to change your request in any way except for writing `https://...` instead of `http://...` into the url when making such a request using a http library/client. The underlying implementation is expected to handle everything else from making the TCP connection through handling the encryption of the data to returning the response.

In practice a http library or client that claims to be able to make "HTTP Requests" is expected to have HTTPS support implemented even if it isn't explicitly documented. Otherwise it would be unusable for most real-world scenarios.

In conclusion: any of the 3 wordings mentioned above in this thread should be correct in this context. Even the plain old "HTTP Request" term.
